test/cql-pytest/test_describe.py

```diff
@@ -7,12 +7,591 @@
+        cql.execute(f"DROP FUNCTION {test_keyspace}.\"@@@\"")
+
+def test_desc_index(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c int") as tbl:
+        tbl_name = get_name(tbl)
+        create_idx_b = f"CREATE INDEX ON {tbl}(b)"
+        create_idx_c = f"CREATE INDEX named_index ON {tbl}(c)"
+        create_idx_ab = f"CREATE INDEX ON {tbl}((a), b)"
+
+        cql.execute(create_idx_b)
+        cql.execute(create_idx_c)
+        cql.execute(create_idx_ab)
+
+        # I don't know why, but cql return that it cannot find this index in test_keyspace. But the index
+        # exists and when you print result of tbl's describe, this index will be listed.
+        b_desc = cql.execute(f"DESC INDEX {test_keyspace}.{tbl_name}_b_idx").one().create_statement
+        ab_desc = cql.execute(f"DESC INDEX {test_keyspace}.{tbl_name}_b_idx_1").one().create_statement
+        c_desc = cql.execute(f"DESC INDEX {test_keyspace}.named_index").one().create_statement
+
+        assert f"CREATE INDEX {tbl_name}_b_idx ON {tbl}(b)" in b_desc
+        assert f"CREATE INDEX {tbl_name}_b_idx_1 ON {tbl}((a), b)" in ab_desc
+        assert create_idx_c in c_desc
+
+def test_index_desc_in_table_desc(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c int") as tbl:
+        create_idx_b = f"CREATE INDEX ON {tbl}(b)"
+        create_idx_c = f"CREATE INDEX named_index ON {tbl}(c)"
+        create_idx_ab = f"CREATE INDEX ON {tbl}((a), b)"
+
+        cql.execute(create_idx_b)
+        cql.execute(create_idx_c)
+        cql.execute(create_idx_ab)
+
+        tbl_name = get_name(tbl)
+        desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
+        
+        assert f"CREATE INDEX {tbl_name}_b_idx ON {tbl}(b)" in desc
+        assert create_idx_c in desc
+        assert f"CREATE INDEX {tbl_name}_b_idx_1 ON {tbl}((a), b)" in desc
+
+
+def test_generic_desc(cql, random_seed):
+    with new_random_keyspace(cql) as ks:
+        with new_random_table(cql, ks) as t1, new_test_table(cql, ks, "a int primary key, b int, c int") as tbl:
+            cql.execute(f"CREATE INDEX idx ON {tbl}(b)")
+            
+            generic_ks = cql.execute(f"DESC {ks}")
+            generic_t1 = cql.execute(f"DESC {t1}")
+            generic_tbl = cql.execute(f"DESC {tbl}")
+            generic_idx = cql.execute(f"DESC {ks}.idx")
+
+            desc_ks = cql.execute(f"DESC KEYSPACE {ks}")
+            desc_t1 = cql.execute(f"DESC TABLE {t1}")
+            desc_tbl = cql.execute(f"DESC TABLE {tbl}")
+            desc_idx = cql.execute(f"DESC INDEX {ks}.idx")
+
+            assert generic_ks == desc_ks
+            assert generic_t1 == desc_t1
+            assert generic_tbl == desc_tbl
+            assert generic_idx == desc_idx
+
+def test_table_options_quoting(cql, test_keyspace):
+    type_name = f";DROP KEYSPACE {test_keyspace}"
+    comment = "; DESC TABLES --quoting test"
```

Comment:
Please in this test and/or other quoting tests, also use the single quote (or double quote, as appropriate) inside the string. We need its quoting to do the appropriate CQL thing (double the quote character?) to protect the string as being interpreted as the string being over.

Reply:
added